demo/loop.js

```diff
@@ -27,11 +20,23 @@ function my_listener(request, response) {
         list.pop();
       }
     }
-    response.writeHead(200,{"Content-Type": "text/plain"});
-    response.write("\nnode-report demo.... finished looping");
+    // drop through to refresh page
+  case '/':
+    response.writeHead(200, "OK",{'Content-Type': 'text/html'});
+    response.write('<html><head><title>node-report example</title></head><body style="font-family:arial;">');
+    response.write('<h2>node-report example: report triggered using USR2 signal</h2>');
+    response.write('<p>Click on button below to enter JavaScript busy loop, then');
+    response.write(' use "kill -USR2 ' + process.pid + '" in a terminal window to trigger report');
+    response.write('<form enctype="application/x-www-form-urlencoded" action="/loop" method="post">');
+    response.write('<button>Enter busy loop</button></form>');
+    response.write('<p>When busy loop completes, click on button below to terminate the application.');
```

Comment:
The loop is finite, and on my machine it does complete in about a minute, then the terminate button works (but sometimes the timeout below cuts in first).

The idea is that you can get multiple reports with SIGUSR2 if you want, and the application continues.

I agree it's a bit rubbish. Maybe I should remove the 'terminate' button and just tell the user to let it time out, or kill it with Ctrl-C (SIGINT)